logrotate.c

```diff
@@ -225,6 +225,33 @@ static int switch_user_back_permanently(void) {
     return ret;
 }
 
+static int open_logfile(const char *path, int flags) {
+    int fd;
+    struct stat sb;
+
+    if ((flags & (O_RDONLY | O_RDWR)) != flags) {
```

Comment:
I do not understand this check.  `flags` is under logrotate's control, isn't it?  Is it meant to be `assert(...)` instead?  Please put there a comment in any case.

Reply:
Refactored the interface of `open_logfile` by not passing any flags at all, just a boolean whether to open with write access.